app/controllers/owners_controller.rb

```diff
@@ -1,6 +1,7 @@
 class OwnersController < ApplicationController
   before_action :find_rubygem, except: :confirm
   before_action :redirect_to_signin, unless: :owner?, except: %i[confirm resend_confirmation]
+  before_action :verify_sudo?, only: %i[index create destroy]
```

Comment:
please make it of the form `before_action :redirect_to_signin, unless: :owner?` and lets not use `sudo`. sudo has special meaning and best not to confuse anyone with fancy terms.
Something like `redirect_to password_verify_path, unless: password_session_expired?`.